pkg/ingester/ingester_v2.go

```diff
@@ -58,11 +62,16 @@ func NewV2(cfg Config, clientConfig client.Config, limits *validation.Overrides,
 		quit:         make(chan struct{}),
 
 		TSDBState: TSDBState{
-			dbs:    make(map[string]*tsdb.DB),
-			bucket: bucketClient,
+			dbs:               make(map[string]*tsdb.DB),
+			bucket:            bucketClient,
+			shipperRegistries: make(map[string]*prometheus.Registry),
 		},
 	}
 
+	if registerer != nil {
+		registerer.MustRegister(i)
```

Comment:
I've the feeling having the `Ingester` to be a `Collector` is not a common pattern. What do you think if we wrap the logic into a custom struct (ie. `shipperMetrics`) which is a `Collector`, while keeping the same design?

Reply:
What do you think about simply using existing `ingesterMetrics` for this?

Reply:
Looks good. Given the quantity of code, may be worth to move it a dedicated file `pkg/ingester/metrics.go`?

Reply:
I actually quite like `shipperMetrics`, because then it's clear that it's TSDB specific and doesn't mix with chunks code.

Reply:
I've moved the code to a new file (incl. ingesterMetrics) and type (`shipperMetrics` now). I've also addressed other feedback. Please take a look.